### ISSUE TITLE
Incorrect package name

### ISSUE THREAD

**Issue by bluefly000** [jroddev/android-oss-release-tracker#3]
This happened when I added the [Aegis authenticator repository](https://github.com/beemdevelopment/Aegis).
The correct package name is "com.beemdevelopment.aegis", but the app recognizes the package name as "me.impy.aegis".

<img src="https://user-images.githubusercontent.com/101441707/201185021-eed3438c-e56c-499a-ac9d-112fcfc24df7.png" width="340px">

**Comment by jroddev** [comment]
@bluefly000 thanks for reporting. I have improved the package detection logic in [0.1.2](https://github.com/jroddev/android-oss-release-tracker/releases/tag/0.1.2).
Aegis now appears to be working as expected.


![image](https://user-images.githubusercontent.com/9654410/201233550-8a51f3d9-5b31-4baa-a90b-1bfb47d9367e.png)

In case you're interested in the details. It was a bug in my defaultBranch detection code. Instead of picking `master` it was selecting the branch: `feature-appbar`
This is the file it was being pulled from https://github.com/beemdevelopment/Aegis/blob/feature-appbar/app/build.gradle

**Comment by bluefly000** [comment]
I had the same problem with other apps. The app version is 0.1.2.

The package name of the application "[Aves](https://github.com/deckerst/aves)" is displayed as "com.android.application". The correct package name is "deckers.thibault.aves".

Also, in the case of [AppManager](https://github.com/MuntashirAkon/AppManager) and [OctoDroid](https://github.com/slapperwan/gh4a), the app could not parse the package name and add the repository.
AppManager package name is "io.github.muntashirakon.AppManager" and OctoDroid package name is "com.gh4a".

<img src="https://user-images.githubusercontent.com/101441707/201396964-8242482e-d730-4a1d-87a6-77bf305c1237.png" width="340px">

**Comment by jroddev** [comment]
@bluefly000 thanks for sharing. I have added support for those 3 in version [0.1.3](https://github.com/jroddev/android-oss-release-tracker/releases/tag/0.1.3)
<img src="https://user-images.githubusercontent.com/9654410/201499802-ce716a98-e1fc-4035-87df-a2f420c69c66.png" width=30% height=30% />

I guess I was wrong about all packageNames having 3 components. These should have actually been picked up by my fallback parser which scrapes the README.md for f-droid links that contain the package name - there was a bug that has now been fixed.

 I have changed the regex to support 2 or more components based on your examples. It uses a non-capturing repeating group and seems to cover the cases I was concerned about 
![image](https://user-images.githubusercontent.com/9654410/201499923-9720e5a6-3c8c-4599-9431-5766277d0a2b.png)
Concerns:
- two component packageNames were conflicting with some variable usages
- including numbers in the packageName was getting confused with some version strings
    
The restrictions I've added are that each component must be two characters or more (letters, numbers and `-` `_`) and that the first component must only be letters. Hopefully this expands the coverage a bit without too many false positives.
